### PR TITLE
Use the new configurator YAML syntax

### LINKED ISSUE #7176: Add docs for the short syntax of service configurators
In https://github.com/symfony/symfony/pull/19190 @voronkovich added support for the short syntax of service configurators. Should we:

1) update the existing examples to always use the short syntax and add the long syntax in a note?
2) Keep existing docs and add the new short syntax in a note?

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #7176.

### CODE REVIEW

File: service_container/configurators.rst [hunk 1]
```diff
@@ -127,12 +127,12 @@ You can configure the service configurator using the ``configurator`` option:
             app.newsletter_manager:
                 class:        AppBundle\Mail\NewsletterManager
                 arguments:    ['@mailer']
-                configurator: ['@app.email_configurator', configure]
+                configurator: 'app.email_configurator:configure'
```
Comment (xabbuh) [inline service_container/configurators.rst]: you actually have to use two colons

**Comment by xabbuh** [comment]
I think we should then document this notation for factories too.

File: service_container/configurators.rst (hunk 1 above)
Comment (javiereguiluz) [inline service_container/configurators.rst]: Like this?

```
configurator: 'app.email_configurator::configure'
```

Comment (stof) [inline service_container/configurators.rst]: @xabbuh that's not true. 2 colons are used when doing a callable referencing a static method (the PHP standard notation ``Class::staticMethod``). The shortcut syntax for service method callables uses a single colon (i.e. something which cannot appear in a valid PHP callable), as done already for the routing configuration.

File: service_container/configurators.rst [hunk 2]
```diff
@@ -186,6 +186,20 @@ You can configure the service configurator using the ``configurator`` option:
             ->setConfigurator(array(new Reference('app.email_configurator'), 'configure'))
         ;
 
+.. note::
```
Comment (stof) [inline service_container/configurators.rst]: shouldn't it be ``versionadded`` ?

File: service_container/configurators.rst (hunk 1 above)
Comment (xabbuh) [inline service_container/configurators.rst]: Sorry for the confusion Javier. @stof is indeed right.

**Comment by javiereguiluz** [comment]
@xabbuh let's do the factories change in another PR: #7208

File: service_container/configurators.rst [hunk 3]
```diff
@@ -186,6 +186,23 @@ You can configure the service configurator using the ``configurator`` option:
...
+
+    .. code-block:: yaml
+
+        app.newsletter_manager:
+            # new syntax
+            configurator: 'app.email_configurator:configure'
+            # old syntax
+            configurator: ['@app.email_configurator', configure]
+
+    .. versionadded:: 3.2
```
Comment (xabbuh) [inline service_container/configurators.rst]: The `versionadded` directive nested in the `note` block doesn't look very nice. What about instead transforming the `note` into a `versionadded` instead as @stof suggested.

Comment (javiereguiluz) [inline service_container/configurators.rst]: Done!

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**